### PR TITLE
Issue 1548: Describe raise_errors better in README

### PR DESCRIPTION
**Pull request by carlwiedemann**
See https://github.com/sinatra/sinatra/issues/1548

The behavior of `raise_errors` isn't well-documented.

When `true`:

* The application will raise errors only for errors that do not have explicit handlers
* The application will raise errors despite any explicit catch-all handler, e.g. `error { ... }` or `error Exception { ... }`

As discussed in https://github.com/sinatra/sinatra/issues/1548, it seems that a future version of sinatra could afford a change to this behavior, since it is somewhat inconsistent, and some options are discussed therein.

In the short-term, we are proposing an update to the documentation.

Once we decide on the final language here, we can open up an accompanying PR for https://github.com/sinatra/sinatra.github.com/ since it [also describes](https://sinatrarb.com/configuration.html) `raise_errors`.

### CODE REVIEW

File: README.md
Comment (dentarg) [inline README.md:2341]: Do we need to preface this with something along the lines "if the environment is `production` or `show_exceptions` is set to `:after_handler`"?

I'm also thinking about "with status code 500", that's correct, but it can also be changed by using `status` inside the handler block, so perhaps it should be made more clear that you are not stuck with 500.

Comment (carlwiedemann) [reply README.md:2341]: Yes, I think there are some nuances here. I'll add that.

Comment (carlwiedemann) [reply README.md:2341]: > I'm also thinking about "with status code 500", that's correct, but it can also be changed by using status inside the handler block, so perhaps it should be made more clear that you are not stuck with 500.

That's true, but I guess what I was trying to describe this specific example, which isn't using the `status` method at all, so I believe it would default to 500. I didn't want to make the example too complicated, because I think this is already a bit complicated to understand.

That said, I'm happy to add some remarks about it, if you believe that it is important.

Comment (carlwiedemann) [reply README.md:2341]: > Do we need to preface this with something along the lines "if the environment is production or show_exceptions is set to :after_handler"?

I attempted this in 44fcd5c5, let me know what you think.

Comment (dentarg) [reply README.md]: > I attempted this in https://github.com/sinatra/sinatra/commit/44fcd5c5a9300e02420593bad808acbddf58f1c0, let me know what you think.

Looks good to me. 👍🏻

>  I guess what I was trying to describe this specific example, which isn't using the status method at all, so I believe it would default to 500. I didn't want to make the example too complicated, because I think this is already a bit complicated to understand.

Agree with that. Could the example be even simpler without mentioning the default 500 status?

I'm fine either way, these are nice improvements to the docs.

Comment (carlwiedemann) [reply README.md]: > Could the example be even simpler without mentioning the default 500 status?

Yes, I think that is a good call. Done in 99ec9445.

**Comment by carlwiedemann** [comment]
Friendly bump. Anything blocking this from getting merged?

**Comment by dentarg** [comment]
I wanted someone else from @sinatra/sinatras-helpers to take a look too, let's see if it happens in the next few days or so

**Review by epergo — APPROVED** [review]
It looks good to me, thanks @carlwiedemann

**Comment by carlwiedemann** [comment]
Thanks @jkowens and @epergo. I don't have write access, so I'll need someone else to merge this if you don't mind. Appreciate it.

**Comment by carlwiedemann** [comment]
Opened https://github.com/sinatra/sinatra.github.com/pull/305 as a follow-up in the docs repo.

**Pull request merged**
